sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -245,10 +415,24 @@ namespace Azure { namespace Core {
       }
     }
 
-    /**
-     * @brief The application context (root).
+    /** @brief Creates a new root Context. This context is not a child of any other context, and
+     * thus can be independently cancelled.
+     *
+     * @return New root context.
+     *
+     */
+    static Context CreateNewRoot() { return Context{std::make_shared<ContextSharedState>()}; }
+
+    /** @brief The `ApplicationContext` is the root of all Context objects.
+     *
+     * @note Cancelling the `ApplicationContext` will cancel ALL operations currently active in the
+     * process, unless they were created with Azure::Core::Context::CreateNewRoot().
+     *
+     * @note If the `ApplicationContext` is cancelled, all current and future operations will be
+     * cancelled. To reset the `ApplicationContext` to a non-cancelled state, call
+     * Azure::Core::Context::ApplicationContext.Reset().
      *
      */
-    static const AZ_CORE_DLLEXPORT Context ApplicationContext;
+    static AZ_CORE_DLLEXPORT Context ApplicationContext;
```

Comment:
I would expect the ApplicationContext to be const with the only affect users can make is to Reset the shared state.

Reply:
If it's const, it cannot be cancelled, and it cannot be reset - that's because both `Cancel` and `Reset` require a mutable Context.

Which throws you into the weird dance you need to perform to cancel the ApplicationContext:

```cpp
auto context = Azure::Core::Context::ApplicationContext;
context.Cancel();
```

If that's the desired dance, then reset can be achieved the same way:
```cpp
auto context = Azure::Core::Context::ApplicationContext;
context.Reset();
```

But I think it's more intuitive to have `ApplicationContext` be mutable, which means you can call:

```cpp
// Cancel the ApplicationContext.
Azure::Core::Context::ApplicationContext.Cancel();
```

I'm ok with the first pattern, but it does seem a bit weird.